tgui/packages/tgui/interfaces/MarkdownViewer.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file MarkdownViewer.tsx
+ * @author ZephyrTFA
+ * @license MIT
+ */
+
+import { Component } from 'inferno';
+import { marked } from 'marked';
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { sanitizeText } from '../sanitize';
+
+type MarkdownViewerData = {
+  title: string;
+  content: string;
+  author: string;
+};
+
+export const MarkdownViewer = (_: any, context: any) => {
+  let { data } = useBackend<MarkdownViewerData>(context);
+  return (
+    <Window theme="paper" title={data.title}>
+      <Window.Content scrollable backgroundColor={'#FFFFFF'}>
+        <MarkdownRenderer content={data.content} />
+      </Window.Content>
+    </Window>
+  );
+};
+
+type MarkdownRendererProps = {
+  content: string;
+  nosanitize?: boolean;
+};
+
+export class MarkdownRenderer extends Component<MarkdownRendererProps> {
+  render() {
+    return <div id="markdown-viewer-content" />;
+  }
+
+  renderMarkdown() {
+    let rendered_content = marked(this.props.content);
+    if (!this.props.nosanitize) {
+      rendered_content = sanitizeText(rendered_content, false);
+    }
+
+    let render_target = document.getElementById('markdown-viewer-content');
+    if (render_target) {
+      render_target.innerHTML = rendered_content;
```

Comment:
Two problems with this method.

1. By using IDs, we can now no longer have two of these at once. You would use refs if you had to.
2. You shouldn't need either of those at all, though. In fact, you don't need a class component here even. This entire thing could be a functional component that sets the `dangerouslySetInnerHTML` property.